src/qililab/instruments/qblox/qblox_qrm.py

```diff
@@ -187,6 +187,33 @@ def acquire_result(self) -> QbloxResult:
         """
         return self.get_acquisitions()
 
+    def acquire_qprogram_results(self, acquisitions: list[str]) -> list[QbloxQProgramMeasurementResult]:  # type: ignore
+        """Read the result from the AWG instrument
+
+        Args:
+            acquisitions (list[str]): A list of acquisitions names.
+
+        Returns:
+            list[QbloxQProgramMeasurementResult]: Acquired Qblox results in chronological order.
+        """
+        return self._get_qprogram_acquisitions(acquisitions=acquisitions)
+
+    @Instrument.CheckDeviceInitialized
+    def _get_qprogram_acquisitions(self, acquisitions: list[str]) -> list[QbloxQProgramMeasurementResult]:
+        results = []
+        for acquisition in acquisitions:
+            for sequencer in self.awg_sequencers:
+                if sequencer.identifier in self.sequences:
```

Comment:
Again, we are not cleaning the cache in `self.sequences`! I don't know if this will bring us problems...

Did you try uploading + executing + getting results of >1 QPrograms in a row? I have the feeling that in the last QPrograms we will also get the results of the first ones...

Reply:
I clean it after acquiring results in the `platform.execute_qprogram` method:

```
# Reset instrument settings
for instrument in self.instruments.elements:
     if isinstance(instrument, QbloxModule):
           instrument.reset_sequences()
           instrument.desync_sequencers()
```

Reply:
Nice! Then we could clean the acquisition data in here aswell (maybe clean it when calling `reset_sequences()`?

Reply:
I added the acquisition data deletion to the `qblox_qrm` since it is relevant only for readout instruments. This loop runs for both control and readouts.